spring-integration-redis/src/main/java/org/springframework/integration/redis/outbound/RedisStoreWritingMessageHandler.java

```diff
@@ -72,22 +69,25 @@
  * @author Oleg Zhurakousky
  * @author Gary Russell
  * @author Mark Fisher
+ * @author Artem Bilan
+ *
  * @since 2.2
  */
 public class RedisStoreWritingMessageHandler extends AbstractMessageHandler {
 
-	private final Log logger = LogFactory.getLog(this.getClass());
-
 	private final Expression zsetIncrementScoreExpression =
-			new SpelExpressionParser().parseExpression("headers." + RedisHeaders.ZSET_INCREMENT_SCORE);
+			new FunctionExpression<Message<?>>(m ->
+					m.getHeaders().get(RedisHeaders.ZSET_INCREMENT_SCORE));
```

Comment:
Not a part of this PR but why is this even an `Expression`? It's not configurable by the user at all.

... time passes ...

>* Fixed logic in RedisCollectionPopulatingMessageHandler to ensure that RedisHeaders.ZSET_INCREMENT_SCORE can be passed as Boolean true/false as well as String true/false. The current solution is based on relyiong oin ConversionService registered with SpEL Context (see RedisCollectionPopulatingMessageHandler.extractZsetIncrementHeader method). This way other types could be supported without introducing any code changes.

Interesting way to achieve that result.

Maybe we should just add a setter to allow the user to specify his own expression.